src/frontc/cparser.mly

```diff
@@ -1365,8 +1415,8 @@ attributes_with_asm:
     /* empty */                         { [] }
 |   attribute attributes_with_asm       { fst $1 :: $2 }
 |   ASM LPAREN string_constant RPAREN attributes
-                                        { ("__asm__",
-					   [CONSTANT(CONST_STRING (fst $3))]) :: $5 }
+                                        { let q,t,l = $3 in ("__asm__",
+					   [CONSTANT(CONST_STRING (queue_to_string q, NO_ENCODING))]) :: $5 }
```

Comment:
Looks like this and the following could still benefit from `string_constant` doing the queue conversion, instead of using the `string_list` renamed to `string_constant`. Or where's the catch?